### ISSUE TITLE
[FEAT] `@interactions.component_callback()` should get the component name from the function name

### ISSUE THREAD

**Issue by Swarkin** [interactions-py/interactions.py#1518]
### Problem Description

similar to how `@interactions.listen` finds the event to listen to or `@interactions.slash_command` the command name from the function itself, `@interactions.component_callback()` should get the component name from the function name.

### Proposed Solution

its easier to type:
```py
@i.component_callback()
async def some_component(...):
    ...
```

instead of:
```py
@i.component_callback('some_component')
async def function(...):
    ...
```

### Code of Conduct

- [X] I agree to follow the contribution requirements.

**Comment by i0bs** [comment]
~~Should be an easy change,~~ I'll try getting this PRed today
edit: I lied.

Changing this may unfortunately result in a breaking change. Our internal code passes an `*args` argument reference, which we cannot assign defaults to needed to allow absence or missing placement. We would either have to change the argument to intake a list, or a singular argument.

Need some lookovers from other org members on this, @AstreaTSS @zevaryx 

edit 2: maybe.

After some discussion with contributors, we may be able to do a workaround with how Python dereferences `*args` as an empty tuple literal of `()`.

**Comment by i0bs** [comment]
Closed in #1519
